paddle/fluid/platform/device/gpu/gpu_launch_config.h

```diff
@@ -30,9 +30,46 @@
 #include <vector>
 #include "paddle/fluid/platform/device_context.h"
 
+#ifdef __HIPCC__
+#define PREDEFINED_BLOCK_SIZE 256
+#else
+#define PREDEFINED_BLOCK_SIZE 512
+#endif
+
 namespace paddle {
 namespace platform {
 
+#ifdef PADDLE_WITH_CUDA
+inline static int RoundToPowerOfTwo(int dim) {
+  if (dim > 512) {
+    return 1024;
+  } else if (dim > 256) {
+    return 512;
+  } else if (dim > 128) {
+    return 256;
+  } else if (dim > 64) {
+    return 128;
+  } else if (dim > 32) {
+    return 64;
+  } else {
+    return 32;
+  }
+}
+#else
+inline static int RoundToPowerOfTwo(int dim) {
+  // HIP results in error or nan if > 256
+  if (dim > 128) {
+    return 256;
+  } else if (dim > 64) {
+    return 128;
+  } else if (dim > 32) {
+    return 64;
+  } else {
+    return 32;
+  }
+}
+#endif
```

Comment:
I think that the working scope of `RoundToPowerOfTwo` is outside `device function`, then I move it from `gpu_device_function.h` header to this seemed more relevant header.